crates/libs/core/src/client/property_client.rs

```diff
@@ -0,0 +1,639 @@
+        timeout_milliseconds: u32,
+        cancellation_token: Option<CancellationToken>,
+    ) -> FabricReceiver<crate::WinResult<()>> {
+        let com1 = &self.com;
+        let com2 = self.com.clone();
+        fabric_begin_end_proxy(
+            move |callback| unsafe {
+                com1.BeginPutPropertyGuid(
+                    name.as_raw(),
+                    property_name.as_pcwstr(),
+                    data,
+                    timeout_milliseconds,
+                    callback,
+                )
+            },
+            move |ctx| unsafe { com2.EndPutPropertyGuid(ctx) },
+            cancellation_token,
+        )
+    }
+
+    fn delete_property_internal(
+        &self,
+        name: &Uri,
+        property_name: &WString,
+        timeout_milliseconds: u32,
+        cancellation_token: Option<CancellationToken>,
+    ) -> FabricReceiver<crate::WinResult<()>> {
+        let com1 = &self.com;
+        let com2 = self.com.clone();
+        fabric_begin_end_proxy(
+            move |callback| unsafe {
+                com1.BeginDeleteProperty(
+                    name.as_raw(),
+                    property_name.as_pcwstr(),
+                    timeout_milliseconds,
+                    callback,
+                )
+            },
+            move |ctx| unsafe { com2.EndDeleteProperty(ctx) },
+            cancellation_token,
+        )
+    }
+
+    fn get_property_metadata_internal(
+        &self,
+        name: &Uri,
+        property_name: &WString,
+        timeout_milliseconds: u32,
+        cancellation_token: Option<CancellationToken>,
+    ) -> FabricReceiver<crate::WinResult<IFabricPropertyMetadataResult>> {
+        let com1 = &self.com;
+        let com2 = self.com.clone();
+        fabric_begin_end_proxy(
+            move |callback| unsafe {
+                com1.BeginGetPropertyMetadata(
+                    name.as_raw(),
+                    property_name.as_pcwstr(),
+                    timeout_milliseconds,
+                    callback,
+                )
+            },
+            move |ctx| unsafe { com2.EndGetPropertyMetadata(ctx) },
+            cancellation_token,
+        )
+    }
+
+    fn get_property_internal(
+        &self,
+        name: &Uri,
+        property_name: &WString,
+        timeout_milliseconds: u32,
+        cancellation_token: Option<CancellationToken>,
+    ) -> FabricReceiver<crate::WinResult<IFabricPropertyValueResult>> {
+        let com1 = &self.com;
+        let com2 = self.com.clone();
+        fabric_begin_end_proxy(
+            move |callback| unsafe {
+                com1.BeginGetProperty(
+                    name.as_raw(),
+                    property_name.as_pcwstr(),
+                    timeout_milliseconds,
+                    callback,
+                )
+            },
+            move |ctx| unsafe { com2.EndGetProperty(ctx) },
+            cancellation_token,
+        )
+    }
+
+    // TODO: implement this
```

Comment:
Is this TODO and the following TODO still relevant?

Reply:
Or the comment is not to implement the code here, but to expose it in PropertyManagementClient ?